package/src/components/Message/utils/messageActions.ts

```diff
@@ -10,21 +10,52 @@ import type {
   UnknownType,
 } from '../../../types/types';
 import type { MessageContextValue } from '../../../contexts/messageContext/MessageContext';
-import type { MessagesContextValue } from '../../../contexts/messagesContext/MessagesContext';
 
-export type MessageActionsProps = {
+export type MessageActionsParams<
+  At extends UnknownType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends UnknownType = DefaultUserType,
+> = {
+  banChannelMembersAllowed: boolean;
   blockUser: MessageActionType;
   copyMessage: MessageActionType;
+  deleteAnyMessageAllowed: boolean;
   deleteMessage: MessageActionType;
+  deleteOwnMessageAllowed: boolean;
+  dismissOverlay: () => void;
   editMessage: MessageActionType;
+  error: boolean | Error;
   flagMessage: MessageActionType;
+  flagMessageAllowed: boolean;
+  isThreadMessage: boolean;
+  messageReactions: boolean;
   muteUser: MessageActionType;
+  muteUserAllowed: boolean;
   pinMessage: MessageActionType;
+  pinMessageAllowed: boolean;
   quotedReply: MessageActionType;
+  quoteMessageAllowed: boolean;
   retry: MessageActionType;
+  sendReplyAllowed: boolean;
   threadReply: MessageActionType;
   unpinMessage: MessageActionType;
-};
+  updateAnyMessageAllowed: boolean;
+  updateOwnMessageAllowed: boolean;
```

Comment:
Sorry if this is obvious and I'm just blanking, but is there a reason to add these here specifically, and not to keep it as one `Capabilities` kind of type connected to the `CapabilitiesContext`? Or "is that the plan for when you get that far?", is probably a better question I guess

Reply:
> Is that the plan for when you get that far

This ^^

Reply:
Mostly playing with capabilities thingy so far

Reply:
Done :)